### PR TITLE
feat(payment): PAYPAL-2611 moved BT venmo button strategy inside packages

### PR DESCRIPTION
**Pull request by andriiVitvitskyi1990**
## What?
Moved BT venmo button strategy inside packages

## Why?
To handle BT venmo as others BT strategies

## Testing / Proof

https://github.com/user-attachments/assets/c6647d0a-d0ca-4faa-8e5c-39eda0183260

https://github.com/user-attachments/assets/5c9783fc-7506-4405-a2fa-26545edcf466

https://github.com/user-attachments/assets/2ad21397-4d81-4e03-a951-1dba0021e672

@bigcommerce/team-checkout @bigcommerce/team-payments

### CODE REVIEW

**Comment by bc-nick** [comment]
@andriiVitvitskyi1990 do we have Venmo button on the PDP?

File: packages/braintree-utils/src/braintree-integration-service.ts [hunk 1]
```diff
@@ -271,6 +274,43 @@ export default class BraintreeIntegrationService {
         return cached;
     }
 
+    async getVenmoCheckout(
```
[the remark this answers is not in the document]
Comment (andriiVitvitskyi1990) [reply packages/braintree-utils/src/braintree-integration-service.ts:277]: done

File: packages/braintree-utils/src/buy-now-cart-request-body.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    optionSelections?: {
+        optionId: number;
+        optionValue: number | string;
+    };
+}
+
+/**
+ * An object that contains the information required for creating 'Buy now' cart.
+ */
+export default interface BuyNowCartRequestBody {
```
[the remark this answers is not in the document]
Comment (andriiVitvitskyi1990) [reply packages/braintree-utils/src/buy-now-cart-request-body.ts]: fixed

File: packages/braintree-utils/src/unsupported-browser-error.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { StandardError } from '@bigcommerce/checkout-sdk/payment-integration-api';
+
+/**
+ * Throw this error if the shopper is using a browser version that is not
+ * supported by us or any third party provider we use.
+ */
+export default class UnsupportedBrowserError extends StandardError {
```
[the remark this answers is not in the document]
Comment (andriiVitvitskyi1990) [reply packages/braintree-utils/src/unsupported-browser-error.ts]: fixed

File: packages/braintree-utils/src/types.ts [hunk 4]
```diff
@@ -48,6 +48,18 @@ export interface BraintreeClient {
     request(payload: BraintreeClientRequestPayload): Promise<BraintreeClientRequestResponse>;
 }
 
+export interface BraintreeDetails {
```
[the remark this answers is not in the document]
Comment (andriiVitvitskyi1990) [reply packages/braintree-utils/src/types.ts]: fixed

**Review by bc-nick — APPROVED** [review]

File: packages/braintree-utils/src/braintree.ts [hunk 5]
```diff
@@ -546,7 +546,7 @@ export type BraintreeVenmoCheckoutCreator = BraintreeModuleCreator<
 >;
 
 export interface BraintreeVenmoCheckout extends BraintreeModule {
-    tokenize(callback: (error: BraintreeError, payload: BraintreeTokenizePayload) => void): void;
+    tokenize(callback: (error: BraintreeError, payload: BraintreeTokenizePayload) => unknown): void;
```
Comment (serhii-tkachenko) [inline packages/braintree-utils/src/braintree.ts:549]: I think it's better to keep `void` here

File: packages/braintree-integration/src/braintree-venmo/braintree-venmo-button-strategy.ts [hunk 6]
```diff
@@ -200,10 +207,10 @@ export default class BraintreeVenmoButtonStrategy implements CheckoutButtonStrat
                         venmoButton.removeAttribute('disabled');
 
                         if (error) {
-                            return this._onError(error);
+                            return this.onError(error);
```
Comment (serhii-tkachenko) [inline packages/braintree-integration/src/braintree-venmo/braintree-venmo-button-strategy.ts:210]: In this case tokenize callback method will return `void`;

```suggestion
                            this.onError(error);
                            
                            return;
```

File: packages/braintree-integration/src/braintree-venmo/braintree-venmo-button-strategy.ts [hunk 7]
```diff
@@ -186,10 +192,11 @@ export default class BraintreeVenmoButtonStrategy implements CheckoutButtonStrat
         venmoButton.setAttribute('aria-label', 'Venmo');
         Object.assign(venmoButton.style, getVenmoButtonStyle(buttonStyles || {}));
 
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
```
Comment (serhii-tkachenko) [inline packages/braintree-integration/src/braintree-venmo/braintree-venmo-button-strategy.ts:195]: Why do you need this line?

**Review by serhii-tkachenko — APPROVED** [review]
Overall looks good to me 👍

File: packages/braintree-integration/src/braintree-venmo/braintree-venmo-button-strategy.ts (hunk 6 above)
Comment (andriiVitvitskyi1990) [reply packages/braintree-integration/src/braintree-venmo/braintree-venmo-button-strategy.ts]: fixed

File: packages/braintree-utils/src/braintree.ts (hunk 5 above)
Comment (andriiVitvitskyi1990) [reply packages/braintree-utils/src/braintree.ts]: fixed

**Review by animesh1987 — APPROVED** [review]

**Pull request merged**